controllers/cluster_controller_phases.go

```diff
@@ -162,6 +163,13 @@ func (r *ClusterReconciler) reconcileInfrastructure(ctx context.Context, cluster
 		return err
 	}
 	cluster.Status.InfrastructureReady = ready
+
+	// Report a summary of current status of the infrastructure object defined for this cluster.
```

Comment:
It looks like we'll return early on line 156 if the infrastructure resource is being deleted. I'm wondering if we need to handle updating conditions in that case as well.

Reply:
That's a good point.
In this first iteration, we are not considering the delete path for conditions, but in the future, I assume that as soon as a controller detects object deletion it should adapt condition accordingly. 
I'm still not sure if we should report such conditions up to the chain, so, if it is ok for you, I will defer this to when I can test the deletion scenario properly